src/backends/wayland/mod.rs

```diff
@@ -263,6 +263,14 @@ pub struct BufferImpl<'a, D: ?Sized, W> {
 }
 
 impl<D: HasDisplayHandle + ?Sized, W: HasWindowHandle> BufferInterface for BufferImpl<'_, D, W> {
+    fn width(&self) -> usize {
+        self.stack.member().width
+    }
+
+    fn height(&self) -> usize {
+        self.stack.member().height
+    }
+
```

Comment:
@ids1024: I'm afraid of touching the `BorrowStack` stuff, maybe I could get you to finish the implementation here?

Reply:
With the `BorrowStack` trick, we can't access both the "container" and "member", so probably it would make the most sense to just copy the `width` and `height` into fields of `BufferImpl`.